src/server/req_quejob.c

```diff
@@ -1177,8 +1178,7 @@ req_quejob(struct batch_request *preq)
 
 			/* unset the old job's JOB_ATR_block */
 			set_jattr_l_slim(pjob, JOB_ATR_block, 0, SET);
-		mark_jattr_not_set(pjob, JOB_ATR_block);
-			pjob->ji_wattr[(int)JOB_ATR_block].at_flags |= ATR_MOD_MCACHE;
+			ATR_UNSET(get_jattr(pj, JOB_ATR_block));
```

Comment:
why this change? we could have just removed the ATR_MOD_MCACHE line below, was this functionally incorrect? If needed, we should change mark_jattr_not_set to call ATR_UNSET